doc/README.md

```diff
@@ -251,25 +251,42 @@ You, as the writer of documentation, are still in charge of its content.
   For example:
 
   ```markdown
-  # pkgs.coolFunction
+  # pkgs.coolFunction {#pkgs.coolFunction}
 
-  Description of what `coolFunction` does.
+  `pkgs.coolFunction` *`name`* *`config`*
 
-  ## Inputs
+  Description of what `callPackage` does.
 
-  `coolFunction` expects a single argument which should be an attribute set, with the following possible attributes:
 
-  `name` (String)
+  ## Inputs {#pkgs-coolFunction-inputs}
+
+  If something's special about `coolFunction`'s general argument handling, you can say so here.
+  Otherwise, just describe the single argument or start the arguments' definition list without introduction.
+
+  *`name`* (String)
 
   : The name of the resulting image.
 
-  `tag` (String; _optional_)
+  *`config`* (Attribute set)
```

Comment:
I also find the current convention `structured function argument` a bit weird. Maybe it is not documented here But its used relatively consistently everywhere. 
Example: 
https://github.com/NixOS/nixpkgs/blob/12cf457ec1bf74dbab75cbeef1f28865d69f6fe4/lib/generators.nix#L131

Do you want to change the convention to `config` (or whatever the actual name is) with `(Attribute Set)`-appendix ?